### PR TITLE
Convert project to Kotlin

### PR DESCRIPTION
**Pull request by KORuL**
* Transferred the project to kotlin
* Removed JavaVersion.VERSION_1_8
* Downgraded target api

### CODE REVIEW

**Comment by KORuL** [comment]
Hi, I needed to use your library in a project where the target is api <26.
To do this, I transferred the project to Kotlin and lowered the api target.
I will be glad to participate in improving the project.
I ask you to check for minor errors, maybe somewhere in the little things I could be mistaken, but I tried to prevent mistakes
I have not yet checked with the devices, since only in 2 days will the glucometer come, for which I did everything.

**Comment by philips77** [comment]
Wow, thanks for the PR. We'll check and test it for sure, but this may take a while.

**Comment by KORuL** [comment]
Fix parseutil - need. I debugged the work with the glucose meter

**Comment by KORuL** [comment]
Do you have a list of supported bluetooth glucometrs?

**Comment by philips77** [comment]
We test only with examples from Nordic SDK: http://developer.nordicsemi.com/nRF5_SDK/nRF5_SDK_v15.x.x/ on nRF5x DevKits. They follow the Bluetooth SIG specification.

File: ble/src/main/java/no/nordicsemi/android/ble/MainThreadBluetoothGattCallback.kt [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+import no.nordicsemi.android.ble.annotation.ConnectionState
+import no.nordicsemi.android.ble.annotation.PhyValue
+
+/**
+ * This class ensures that the BLE callbacks will be called on the main (UI) thread.
+ * Handler parameter was added to [ #connectGatt(Context, boolean, BluetoothGattCallback, int, int, Handler)][android.bluetooth.BluetoothDevice]
+ * in Android Oreo, before that the behavior was undefined.
+ */
+abstract class MainThreadBluetoothGattCallback : BluetoothGattCallback() {
+    private var mHandler: Handler? = Handler()
```
Comment (keletappi) [inline ble/src/main/java/no/nordicsemi/android/ble/MainThreadBluetoothGattCallback.kt]:
```suggestion
    private var mHandler: Handler? = Handler(Looper.getMainLooper())
```

Argumentless Handler() constructor uses Looper from currently running thread. This means that if this object is created from a background thread, the runnables posted to the handler are executed on that thread, not main thread.

Comment (philips77) [reply ble/src/main/java/no/nordicsemi/android/ble/MainThreadBluetoothGattCallback.kt]: Very good observation, thank you.

**Comment by philips77** [comment]
I have a problem with migration BLE Library to Kotlin. I'm aware of few bugs in the latest version and it would be much easier for me to fix them in Java than in Kotlin, as I would have to learn everything.

**Comment by KORuL** [comment]
Thank you very much for the review!

**Comment by KORuL** [comment]
You can show your edits in java in some way, and I will transfer them to kotlin.

**Comment by philips77** [comment]
I'm sorry, but this PR got obsolete now.

**Pull request closed without merging**